specification/network/resource-manager/Microsoft.Network/stable/2019-09-01/bastionHost.json

```diff
@@ -290,6 +262,225 @@
           "nextLinkName": "nextLink"
         }
       }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/bastionHosts/{bastionHostName}/createbsl": {
+      "post": {
+        "tags": [
+          "BastionHosts",
+          "BastionShareableLink"
+        ],
+        "operationId": "PutBastionShareableLink",
+        "description": "Creates a Bastion Shareable Links for all the VMs specified in the request.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ResourceGroupName"
+          },
+          {
+            "$ref": "#/parameters/BastionHostName"
+          },
+          {
+            "$ref": "./network.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "./network.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/BastionShareableLinkRequest"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success. The operation returns a list of shareable links for the specified VMs, giving appropriate error messages as needed.",
+            "schema": {
+              "$ref": "#/definitions/BastionShareableLinkListResult"
+            }
+          },
+          "202": {
+            "description": "Accepted and the operation will complete asynchronously."
+          },
+          "400": {
+            "description": "Bad request. Check the message for details."
+          },
+          "404": {
+            "description": "No access to the bastion or one of the requested virtual machines."
+          }
+        },
+        "x-ms-examples": {
+          "Create Bastion Shareable Links for the request VMs": {
+            "$ref": "./examples/BastionShareableLinkCreate.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/bastionHosts/{bastionHostName}/deletebsl": {
+      "post": {
```

Comment:
Why do you use 'post' for delete and get operations?

Reply:
These APIs are used for the used for creating/deleting/fetching the shareable links or the sessions that are currently on the Bastion. However, the shareable links or the sessions don't exist as a separate resource in Azure. So we used the post operations for them

Reply:
By adding these APIs to your RP, sharable links/sessions would exist in azure and would need to be modeled appropriately.

*bsl should be modeled as a CRUD-able resource, not using post actions. Using actions like this changes the required permissions and API expectations of the end user, breaking consistency requirements of ARM apis.

The name of the resource should be spelled out too: 'sharableLinks'.

Reply:
I don't see an answer to Ryan's question about modeling this as a CRUD-able resource instead of a bunch of POST actions.